SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/UserIdContext.cs

```diff
@@ -48,7 +57,27 @@ public static UserIdContext Instance
         /// <summary>
         /// Current user identifier.
         /// </summary>
-        public string UserId { get; set; }
+        public string UserId
+        {
+            get
+            {
+                lock (UserIdContextLock)
+                {
+                    return _userId;
+                }
+            }
+            set
+            {
+                lock (UserIdContextLock)
+                {
+                    if (_userId == null || !_userId.Equals(value))
```

Comment:
If you update from null to null, that's not a change but this will trigger the event here.
Make sure to check for that and add a test.

Reply:
```suggestion
                    if (_userId != value)
```

Would this work?